controllers/articlesController.js

```diff
@@ -311,4 +284,145 @@ export default class PostController {
       });
     }
   }
+  /**
+   * @description It helps the user to fetch all of the created articles.
+   * @param  {object} req - The request object
+   * @param  {object} res - The response object
+   * @returns {object} It returns the request's response object
+   */
+
+  static async findAll(req, res) {
+    const { user: { id = null } = {} } = req;
+    const { page = 1, limit = 20 } = req.query;
+    try {
+      const all = await Article.findAll({
+        offset: (Number(page) - 1) * Number(limit),
+        limit,
+        order: [["createdAt", "DESC"]],
+        include: [
+          {
+            model: User,
+            as: "author",
+            attributes: ["username", "bio", "image"],
+            include: [
+              { model: User, as: "followers", attributes: ["id", "username"] }
+            ]
+          },
+          { model: User, as: "likes", attributes: ["username"], through: {} },
+          { model: Tag, attributes: ["name"], as: "tagsList", through: {} }
+        ]
+      });
+      if (all.length) {
+        const articles = all.map(each => {
+          const {
+            author: { followers, username, bio, image }
+          } = each.get();
+          const { tagsList, likes, ...article } = each.get();
+          const following = !!_.dropWhile(followers, f => f.id !== id).length;
+          return {
+            ...article,
+            likes: likes.map(like => like.username),
+            author: { following, username, bio, image },
+            favorited: !!likes.length,
+            favoritesCount: likes.length,
+            tagsList: tagsList.map(t => t.get().name || null)
+          };
+        });
+        const articlesCount = articles.length;
+        return res
+          .status(OK)
+          .json({ message: "Successful", articles, articlesCount });
+      }
+      return res.status(NOT_FOUND).json({
+        message: "No more articles found, still need to read more?"
```

Comment:
Why are you asking "if they still need to read more", is they a way user will get chance to provide "yes" or "no" as an answer?

Reply:
This message tells the user that all articles were fetched, as the title of the PR states, this is pagination. It means there will be a time when the user is scrolling through and the app is done fetching all articles. So, it will tell the user that all there are no more articles. There is no way the user will respond, let me tweak it a bit.